homeassistant/components/sensor/season.py

```diff
@@ -0,0 +1,149 @@
+STATE_WINTER = 'Winter'
+TYPE_ASTRONOMICAL = 'astronomical'
+TYPE_METEOROLOGICAL = 'meteorological'
+VALID_TYPES = [TYPE_ASTRONOMICAL, TYPE_METEOROLOGICAL]
+
+HEMISPHERE_SEASON_SWAP = {STATE_WINTER: STATE_SUMMER,
+                          STATE_SPRING: STATE_AUTUMN,
+                          STATE_AUTUMN: STATE_SPRING,
+                          STATE_SUMMER: STATE_WINTER}
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_TYPE, default=TYPE_ASTRONOMICAL): vol.In(VALID_TYPES)
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Display the current season."""
+    if None in (hass.config.latitude, hass.config.longitude):
+        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
+        return False
+
+    latitude = util.convert(hass.config.latitude, float)
+    _type = config.get(CONF_TYPE)
+
+    if latitude < 0:
+        hemisphere = SOUTHERN
+    elif latitude > 0:
+        hemisphere = NORTHERN
+    else:
+        hemisphere = EQUATOR
+
+    _LOGGER.debug(_type)
+    add_devices([Season(hass, hemisphere, _type)])
+
+    return True
+
+
+class Season(Entity):
+    """Representation of the current season."""
+
+    def __init__(self, hass, hemisphere, season_tracking_type):
+        """Initialize the season."""
+        self.hass = hass
+        self.hemisphere = hemisphere
+        self.datetime = datetime.now()
+        self.type = season_tracking_type
+        self.season = self.get_season()
+
+    @property
+    def name(self):
+        """Return the name."""
+        return "Season"
+
+    @property
+    def state(self):
+        """Return the current season."""
+        return self.season
+
+    def get_season(self):
+        """Calculate the current season."""
+        if self.hemisphere == 'equator':
+            return STATE_UNKNOWN
+
+        spring_start = self.spring_start()
+        summer_start = self.summer_start()
+        autumn_start = self.autumn_start()
+        winter_start = self.winter_start()
+
+        if spring_start <= self.datetime < summer_start:
+            season = STATE_SPRING
+        elif summer_start <= self.datetime < autumn_start:
+            season = STATE_SUMMER
+        elif autumn_start <= self.datetime < winter_start:
+            season = STATE_AUTUMN
+        elif winter_start <= self.datetime or spring_start > self.datetime:
+            season = STATE_WINTER
+
+        # If user is located in the southern hemisphere swap the season
+        if self.hemisphere == NORTHERN:
+            return season
+        return HEMISPHERE_SEASON_SWAP.get(season)
+
+    def spring_start(self):
+        """The date of the first day of spring."""
+        if self.type == TYPE_ASTRONOMICAL:
+            import ephem
+            return ephem.next_equinox(str(self.datetime.year)).datetime()
+        year = str(self.datetime.year)
+        start_of_spring = "03/01/" + year
+        return datetime.strptime(start_of_spring, '%m/%d/%Y')
```

Comment:
It would be faster to create datetimes and just replace the year instead of going via strings

![image](https://user-images.githubusercontent.com/1444314/29600757-a51868aa-878c-11e7-8626-62c8b1f1da8e.png)